webpack/assets/javascripts/react_app/components/AutoCompleteSearch/helpers.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Menu, MenuItem, Highlighter } from 'react-bootstrap-typeahead';
+
+export const groupBy = (list, keyGetter) => {
+  const map = new Map();
+  list.forEach((item) => {
+    const key = keyGetter(item);
+    const collection = map.get(key);
+    if (!collection) {
+      map.set(key, [item]);
+    } else {
+      collection.push(item);
+    }
+  });
+  return map;
+};
+
+export const renderMenu = (results, menuProps) => {
+  let idx = 0;
+  const grouped = groupBy(results, r => r.category);
+  const items = [...grouped.entries()].sort().map(category => [
+    !!idx && <Menu.Divider key={`${category}-divider`} />,
+      <Menu.Header key={`${category}-header`}>
+        {category[0]}
+      </Menu.Header>,
+      category[1].map((option) => {
+        const item =
+          <MenuItem key={idx} option={option.label} position={idx}>
+            <Highlighter search={menuProps.text}>
```

Comment:
There is a component that probably does the same:
https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/SubstringWrapper/index.js#L4
we can use `Highlighter` instead in other places